### DISCUSSION TITLE
Get a CDSView from the current plot state

### DISCUSSION THREAD

**Discussion by aprsa** [bokeh/bokeh#14181] [Internals and design (Q&A)]
I understand how to filter the data to plot a subset using CDSView. What I would need is the opposite: to zoom in on a particular region of the plot and use that as filter that would be captured in a CDSView. Of course I could probe x_range and y_range and then form a filter from that and generate a CDSView, but I'd like to avoid that if possible. It intuitively seems to me that bokeh should have a CDSView-like concept of what is being currently displayed. Can that be done?

**Comment by mattpap** [comment]
BokehJS implements `CDSView.masked` indices which includes indices contained in the viewport, but that's an internal property and it only works for a subset of glyphs. There has been some effort to improve this situation, e.g. in PR #13332.
